openfl/component/director/director.py

```diff
@@ -64,20 +64,25 @@ async def set_new_experiment(self, *, experiment_name: str, sender_name: str,
         """Set new experiment."""
         # TODO: save to file
         self.experiment_data[experiment_name] = data
+        experiment_work_dir = Path(f'{os.getcwd()}/{sender_name}/{experiment_name}')
 
-        self.create_workspace(experiment_name, data)
+        self.create_workspace(experiment_name, experiment_work_dir, data)
         asyncio.create_task(self._run_aggregator(
-            sender_name,
-            tensor_dict,
-            experiment_name,
-            collaborator_names
+            experiment_sender=sender_name,
+            initial_tensor_dict=tensor_dict,
+            experiment_name=experiment_name,
+            collaborator_names=collaborator_names,
+            experiment_work_dir=experiment_work_dir
         ))
 
         logger.info(f'New experiment {experiment_name} for '
                     f'collaborators {collaborator_names}')
         for col_name in collaborator_names:
             queue = self.col_exp_queues[col_name]
-            await queue.put(experiment_name)
+            await queue.put({
+                'experiment_name': experiment_name,
+                'client_id': sender_name
```

Comment:
Don't we disclose the experiment owner's identity to envoys here? Is it necessary?